app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -364,26 +368,59 @@ public int onStartCommand(Intent intent, int flags, int startId) {
         return START_STICKY;
     }
 
-    private void startPeriodicScreenScan() {
+    private void watchScreen() {
         area[0] = new Point(                // these values used to get "white" left of "power up"
                 Math.round(displayMetrics.widthPixels / 24),
                 (int) Math.round(displayMetrics.heightPixels / 1.24271845));
         area[1] = new Point(                // these values used to get greenish color in transfer button
                 (int) Math.round(displayMetrics.widthPixels / 1.15942029),
                 (int) Math.round(displayMetrics.heightPixels / 1.11062907));
-        final Handler handler = new Handler();
-        timer = new Timer();
-        TimerTask doAsynchronousTask = new TimerTask() {
+
+        screenScanHandler = new Handler();
+        screenScanRunnable = new Runnable() {
             @Override
             public void run() {
-                handler.post(new Runnable() {
-                    public void run() {
-                        scanPokemonScreen();
-                    }
-                });
+                if (screenScanRetries > 0) {
+                    scanPokemonScreen();
+                    screenScanRetries--;
+                    screenScanHandler.postDelayed(screenScanRunnable, 500);
```

Comment:
I hadn't realized we could have used `postDelayed` instead of `TimerTask` *all this time* 😢